src/software/thunderscope/replay/replay_file_debugging_script.py

```diff
@@ -0,0 +1,106 @@
+import traceback
+import base64
+import glob
+import gzip
+import os
+import argparse
+
+from proto.import_all_protos import *
+from software.thunderscope.replay.replay_constants import *
+
+
+def read_one_chunk(replay_file_name):
+    """
+    read one chunk of the replay file:
+
+    :param replay_file_name: the name of the replay file that is going to be read
+    :return: the number of lines that have been read
+    """
+
+    line_num = 0
+    with gzip.open(replay_file_name, "rb") as replay_file:
+
+        for line in replay_file.readlines():
+            # do not parse empty line
+            if line == b"":
+                continue
+
+            try:
+                timestamp, protobuf_type, data = line.split(
+                    bytes(REPLAY_METADATA_DELIMETER, encoding="utf-8")
+                )
+            except Exception as e:
+                print("Exception ignored. Please see below for more!")
+                print(e)
+                continue
+
+            try:
+                # The format of the protobuf type is:
+                # package.proto_class (e.g. TbotsProto.Primitive)
+                proto_class = eval(str(protobuf_type.split(b".")[-1], encoding="utf-8"))
+            except Exception as e:
+                print("Error ignored with Exception")
+                print(e)
+                continue
+
+            # Deserialize protobuf
+            proto = proto_class.FromString(
+                base64.b64decode(data[len("b") : -len("\n")])
+            )
+
+            #######################################
+            # Do something with the protobuf here #
+            #######################################
+            print(
+                "{}: {}: {} - {=}".format(
+                    line_num, float(timestamp), protobuf_type, proto
+                )
+            )
```

Comment:
This doesn't run right now since `{=}` doesn't seem to be supported in str.format
```suggestion
            print(
                "{}: {}: {} - {}".format(
                    line_num, float(timestamp), protobuf_type, proto
                )
            )
```